libc/spec/spec.td

```diff
@@ -169,10 +169,30 @@ class ArgSpec<Type type, list<Annotation> annotations = [], string name = ""> {
   string Name = name;
 }
 
-class FunctionSpec<string name, RetValSpec return, list<ArgSpec> args> {
+class FunctionAttr {}
+class GnuFunctionAttr<string attr> : FunctionAttr {
+  string Attr = attr;
+  string Style = "gnu";
+}
+class Cxx11FunctionAttr<string attr, string namespace> : FunctionAttr {
+  string Attr = attr;
+  string Namespace = namespace;
+  string Style = "cxx11";
+}
+class DeclspecFunctionAttr<string attr> : FunctionAttr {
+  string Attr = attr;
+  string Style = "declspec";
+}
+class FunctionAttrSpec<string macro, list<FunctionAttr> instances> {
+  list<FunctionAttr> Instances = instances;
+  string Macro = macro;
+}
```

Comment:
Besides function level attributes, there are also parameter level attributes.  Less likely to use those, but worth considering for the future.

Reply:
We already kind of do that ad-hoc with `__restrict`, might be good to formalize.

Reply:
`__restrict` is a qualifier though, not a parameter attribute. Qualifiers are `restrict`, `const`, and `volatile`.

Reply:
Mentioned in comment.